client/src/app.tsx

```diff
@@ -1,69 +1,108 @@
-import React, {Component, Fragment, ReactNode} from 'react';
-import Menu from './components/menu';
-import {Notifier} from './components/notifier';
-import Error from './components/error';
-import {initRouter} from './router';
-import log from './utils/log';
-import Button from './components/button';
-import LogoSvg from './art/k8dashSvg';
-import HamburgerSvg from './art/hamburgerSvg';
+import React, { Component, Fragment, ReactNode } from "react";
+import Menu from "./components/menu";
+import { Notifier } from "./components/notifier";
+import Error from "./components/error";
+import { initRouter } from "./router";
+import log from "./utils/log";
+import Button from "./components/button";
+import LogoSvg from "./art/k8dashSvg";
+import HamburgerSvg from "./art/hamburgerSvg";
 
 type State = {
-    content?: ReactNode;
-    contentDate?: number;
-    hasError?: boolean;
-    menuToggled?: boolean;
-}
+  content?: ReactNode;
+  contentDate?: number;
+  hasError?: boolean;
+  menuToggled?: boolean;
+};
 
 const genericError = (
-    <div id='content'>
-        <Error messages={['There was an error and this content was unable to be displayed.']} />
-    </div>
+  <div id="content">
+    <Error
+      messages={[
+        "There was an error and this content was unable to be displayed.",
+      ]}
+    />
+  </div>
 );
 
 class App extends Component<{}, State> {
-    componentDidMount() {
-        initRouter((content: ReactNode) => {
-            this.setState({content, contentDate: Date.now(), hasError: false});
-            window.scrollTo(0, 0);
-        });
+  componentDidMount() {
+    initRouter((content: ReactNode) => {
+      this.setState({ content, contentDate: Date.now(), hasError: false });
+      window.scrollTo(0, 0);
+    });
+    this.detectColorScheme();
+  }
+
+  detectColorScheme = () => {
+    let theme = "light"; //default to light
+
+    //local storage is used to override OS theme settings
+    if (localStorage.useDarkMode) {
+      if (localStorage.useDarkMode === "true") {
+        theme = "dark";
+      }
+    } else if (!window.matchMedia) {
+      //matchMedia method not supported
+      return false;
+    } else if (window.matchMedia("(prefers-color-scheme: dark)").matches) {
+      //OS theme setting detected as dark
+      theme = "dark";
+      localStorage.useDarkMode = "true";
```

Comment:
Does this cause the matchMedia based selection to be ignored in the future if the user switches their browser to a light theme?

Reply:
I think this logic should move into `account.tsx` to keep everything in one place. Then, I think the way this should work is...

1. If the user has manually selected light/dark mode, use their selection (for this to work, we'll need to change the logic in `setDarkMode` to not delete `localSorage.useDarkMode`
2. If they haven't then test `window.matchMedia("(prefers-color-scheme: dark)").matches` and set dark mode based on that (this should be a small change in `setDarkModeClass` I think)

This does mean that once a user manually sets dark/light mode, there's no way for them to go back to using `prefers-color-scheme`, but I think that's ok for now. If someone complains, we can just let them know to clear out the setting in localStorage. If it becomes an on-going issue, we'll figure out how to deal with it :)

Reply:
Maybe the user preferences could offer an option to use system defaults that simply deletes the setting from localStorage for them.